File: Backend/FilmHarbor.Solution/FilmHarbor.Core/Entities/Movie.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FilmHarbor.Core.Entities
{
    public class Movie
    {
        public int Id { get; set; }

        [StringLength(30)]
        public string Title { get; set; } = null!;

        public int CategoryId { get; set; }

        public int? ReleaseYear { get; set; }

        [StringLength(255)]
        public string? ImageUrl { get; set; }

        [StringLength(100)]
        public string? Description { get; set; }


        //Navigattion Properties
        public Category? Category { get; set; }

        public ICollection<Review> Reviews { get; set; } = new List<Review>();

        public ICollection<User> FavouriteByUsers { get; set; } = new List<User>();
    }
}

File: Frontend/FilmHarborApp/src/app/models/movie.ts
import { Category } from './category';

export interface Movie {
  id: number;
  title: string;
  categoryId: number;
  releaseYear: number | null;
  imageUrl: string | null;
  description: string | null;
  category: Category | null;
  // reviews: Review[];
  // favouriteByUsers: User[];
}
